File: api/convert.py
"""
Vercel Serverless Function - POST /convert
Converts Java Selenium code to Playwright TypeScript
"""
import json

def get_mock_conversion(java_code: str) -> str:
    """Returns a mock Playwright conversion for demo purposes."""
    return '''import { test, expect } from '@playwright/test';

test('converted test', async ({ page }) => {
    // Navigate to the page
    await page.goto('https://example.com');
    
    // Fill in form fields
    await page.locator('#username').fill('user');
    await page.locator('#password').fill('pass');
    
    // Click submit button
    await page.locator('#submit').click();
    
    // Verify page title
    await expect(page).toHaveTitle('Dashboard');
});

// NOTE: This is a DEMO conversion.
// To get real AI-powered conversions, configure a valid API key in Vercel Environment Variables.'''

def handler(event, context):
    """AWS Lambda / Vercel serverless handler"""
    method = event.get('httpMethod', 'POST')
    
    headers = {
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
    }
    
    if method == 'OPTIONS':
        return {
            'statusCode': 200,
            'headers': {**headers, 'Access-Control-Allow-Methods': 'POST, OPTIONS'},
            'body': json.dumps({'message': 'OK'})
        }
    
    if method == 'POST':
        try:
            body_str = event.get('body', '{}')
            if isinstance(body_str, str):
                body = json.loads(body_str)
            else:
                body = body_str
            
            java_code = body.get('java_source_code', '')
            
            if not java_code:
                return {
                    'statusCode': 400,
                    'headers': headers,
                    'body': json.dumps({
                        'playwright_code': '',
                        'status': 'error',
                        'error_message': 'No Java code provided'
                    })
                }
            
            result = get_mock_conversion(java_code)
            
            return {
                'statusCode': 200,
                'headers': headers,
                'body': json.dumps({
                    'playwright_code': result,
                    'status': 'success',
                    'error_message': None
                })
            }
            
        except Exception as e:
            return {
                'statusCode': 500,
                'headers': headers,
                'body': json.dumps({
                    'playwright_code': '',
                    'status': 'error',
                    'error_message': str(e)
                })
            }
    
    return {
        'statusCode': 405,
        'headers': headers,
        'body': json.dumps({'error': 'Method not allowed'})
    }

File: api/convert.js
export const config = {
  runtime: 'edge',
};

const SYSTEM_PROMPT = `You are a code converter expert. Convert Java Selenium WebDriver code to Playwright TypeScript.

Rules:
1. Convert TestNG @Test annotations to Playwright test()
2. Replace WebDriver with Playwright page object
3. Use Playwright's locator() instead of findElement()
4. Use async/await for all Playwright operations
5. Import { test, expect } from '@playwright/test'
6. Remove WebDriver setup/teardown - use Playwright's built-in fixtures
7. Convert Assert.assertEquals() to expect().toBe() or expect().toEqual()

Output only the TypeScript code, no explanations.`;

export default async function handler(request) {
  // Enable CORS
  if (request.method === 'OPTIONS') {
    return new Response(null, {
      status: 200,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Methods': 'POST, OPTIONS',
        'Access-Control-Allow-Headers': 'Content-Type',
      },
    });
  }

  if (request.method !== 'POST') {
    return new Response(JSON.stringify({ error: 'Method not allowed' }), {
      status: 405,
      headers: { 'Content-Type': 'application/json' },
    });
  }

  try {
    const body = await request.json();
    const java_code = body.java_source_code;

    if (!java_code) {
      return new Response(
        JSON.stringify({
          playwright_code: '',
          status: 'error',
          error_message: 'No Java code provided',
        }),
        {
          status: 400,
          headers: {
            'Content-Type': 'application/json',
            'Access-Control-Allow-Origin': '*',
          },
        }
      );
    }

    // Call Groq API
    const apiKey = process.env.GROQ_API_KEY;
    
    if (!apiKey) {
      return new Response(
        JSON.stringify({
          playwright_code: '',
          status: 'error',
          error_message: 'API key not configured',
        }),
        {
          status: 500,
          headers: {
            'Content-Type': 'application/json',
            'Access-Control-Allow-Origin': '*',
          },
        }
      );
    }

    const response = await fetch('https://api.groq.com/openai/v1/chat/completions', {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${apiKey}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        model: 'llama-3.1-8b-instant',
        messages: [
          { role: 'system', content: SYSTEM_PROMPT },
          { role: 'user', content: `Convert this Java Selenium code to Playwright TypeScript:\n\n${java_code}` }
        ],
        temperature: 0.2,
        max_tokens: 4096,
      }),
    });

    if (!response.ok) {
      const error = await response.text();
      console.error('Groq API error:', error);
      return new Response(
        JSON.stringify({
          playwright_code: '',
          status: 'error',
          error_message: `AI service error: ${response.status}`,
        }),
        {
          status: 500,
          headers: {
            'Content-Type': 'application/json',
            'Access-Control-Allow-Origin': '*',
          },
        }
      );
    }

    const data = await response.json();
    const convertedCode = data.choices?.[0]?.message?.content || '';

    return new Response(
      JSON.stringify({
        playwright_code: convertedCode,
        status: 'success',
        error_message: null,
      }),
      {
        status: 200,
        headers: {
          'Content-Type': 'application/json',
          'Access-Control-Allow-Origin': '*',
        },
      }
    );
  } catch (error) {
    console.error('Conversion error:', error);
    return new Response(
      JSON.stringify({
        playwright_code: '',
        status: 'error',
        error_message: error.message,
      }),
      {
        status: 500,
        headers: {
          'Content-Type': 'application/json',
          'Access-Control-Allow-Origin': '*',
        },
      }
    );
  }
}
